### PR TITLE
WebXr updates

### PR DESCRIPTION
**Pull request by slimbuck**
This PR:
- add shadow catcher
- update xr mode as follows:
    - place the object infront of user instead of at user's feet
    - support tap-to-place
    - use light estimation
- add new lighting environments
- set any parameter from url search param
- update to engine v1.64.3

### CODE REVIEW

File: src/shadow-catcher.ts [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+    Layer,
+    MeshInstance,
+    RenderComponent,
+    StandardMaterial
+} from 'playcanvas';
+
+const endPS = `
+    litShaderArgs.opacity = mix(light0_shadowIntensity, 0.0, shadow0);
+    gl_FragColor.rgb = vec3(0.0);
+    // gl_FragColor.rgb = vec3(mix(light0_shadowIntensity, 0.0, shadow0));
```
Comment (willeastcott) [inline src/shadow-catcher.ts:19]: Delete this line?

File: src/shadow-catcher.ts [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+
+        const layers = app.scene.layers;
+        const worldLayer = layers.getLayerByName('World');
+        const idx = layers.getTransparentIndex(worldLayer);
+        layers.insert(this.layer, idx + 1);
+
+        // create shadow catcher material
+        this.material = new StandardMaterial();
+        this.material.useSkybox = false;
+        this.material.blendType = BLEND_NORMAL;
```
Comment (willeastcott) [inline src/shadow-catcher.ts:44]: In the WebXR Starter Kit, it uses BLEND_PREMULTIPLIED. Which is right?

File: src/shadow-catcher.ts [hunk 3]
```diff
@@ -0,0 +1,132 @@
+import {
+    BLEND_NORMAL,
+    SHADOW_VSM16 as SHADOW_TYPE,
+    SHADOWUPDATE_REALTIME as SHADOWUPDATE,
```
Comment (willeastcott) [inline src/shadow-catcher.ts:4]: Maybe it's just me, but I found this use of `as` confusing when I read the light component creation code.

**Review by willeastcott — APPROVED** [review]
Awesome improvements!

File: src/shadow-catcher.ts (hunk 1 above)
Comment (slimbuck) [reply src/shadow-catcher.ts]: done

File: src/shadow-catcher.ts (hunk 2 above)
Comment (slimbuck) [reply src/shadow-catcher.ts:43]: If shadow is black then they're equivalent.

**Pull request merged**